lib/mix/tasks/phx.gen.auth.ex

```diff
@@ -222,38 +224,69 @@ defmodule Mix.Tasks.Phx.Gen.Auth do
-      {:eex, "reset_password_controller_test.exs",
-       Path.join([web_test_prefix, "controllers", web_path, "#{schema.singular}_reset_password_controller_test.exs"])},
-      {:eex, "reset_password_edit.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_reset_password", "edit.html.heex"])},
-      {:eex, "reset_password_new.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_reset_password", "new.html.heex"])},
-      {:eex, "session_view.ex", Path.join([web_prefix, "views", web_path, "#{schema.singular}_session_view.ex"])},
-      {:eex, "session_controller.ex", Path.join([web_prefix, "controllers", web_path, "#{schema.singular}_session_controller.ex"])},
-      {:eex, "session_controller_test.exs", Path.join([web_test_prefix, "controllers", web_path, "#{schema.singular}_session_controller_test.exs"])},
-      {:eex, "session_new.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_session", "new.html.heex"])},
-      {:eex, "settings_view.ex", Path.join([web_prefix, "views", web_path, "#{schema.singular}_settings_view.ex"])},
-      {:eex, "settings_edit.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_settings", "edit.html.heex"])},
-      {:eex, "settings_controller.ex", Path.join([web_prefix, "controllers", web_path, "#{schema.singular}_settings_controller.ex"])},
-      {:eex, "settings_controller_test.exs", Path.join([web_test_prefix, "controllers", web_path, "#{schema.singular}_settings_controller_test.exs"])}
-    ]
+    case Keyword.fetch(context.opts, :live) do
+      {:ok, true} ->
+        [
+          {:eex, "migration.ex", Path.join([migrations_prefix, "#{timestamp()}_create_#{schema.table}_auth_tables.exs"])},
+          {:eex, "notifier.ex", Path.join([context.dir, "#{schema.singular}_notifier.ex"])},
+          {:eex, "schema.ex", Path.join([context.dir, "#{schema.singular}.ex"])},
+          {:eex, "schema_token.ex", Path.join([context.dir, "#{schema.singular}_token.ex"])},
+          {:eex, "auth.ex", Path.join([web_prefix, web_path, "#{schema.singular}_auth.ex"])},
+          {:eex, "auth_test.exs", Path.join([web_test_prefix, web_path, "#{schema.singular}_auth_test.exs"])},
+          {:eex, "confirmation_view.ex", Path.join([web_prefix, "views", web_path, "#{schema.singular}_confirmation_view.ex"])},
+          {:eex, "confirmation_new.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_confirmation", "new.html.heex"])},
+          {:eex, "confirmation_edit.html.heex", Path.join([web_prefix, "templates", web_path, "#{schema.singular}_confirmation", "edit.html.heex"])},
+          {:eex, "confirmation_controller.ex", Path.join([web_prefix, "controllers", web_path, "#{schema.singular}_confirmation_controller.ex"])},
+          {:eex, "confirmation_controller_test.exs", Path.join([web_test_prefix, "controllers", web_path, "#{schema.singular}_confirmation_controller_test.exs"])},
```

Comment:
I see that some of these are going to be present regardless of the `--live` option, if I'm not overseeing it. So, would it make sense to extract the default and concat only the diff?

```elixir
defaults = [
 ...
]

case Keyword.fetch(context.opts, :live) do
  {:ok, true} ->
    live_files = [...]
    defaults ++ live_files
    
  _ ->
    non_live_files = [...]
    defaults ++ non_live_files
end

```

Reply:
Make sense! I was wondering how it was easier to read the entire list of routes for each case.